app/src/main/AndroidManifest.xml

```diff
@@ -135,189 +134,632 @@
+                <data android:host="videos.hauspie.fr" />
+                <data android:host="videos.iut-orsay.fr" />
+                <data android:host="videos.koumoul.com" />
+                <data android:host="videos.koweb.fr" />
+                <data android:host="videos.lavoixdessansvoix.org" />
+                <data android:host="videos.lescommuns.org" />
+                <data android:host="videos-libr.es" />
+                <data android:host="videos.numerique-en-commun.fr" />
+                <data android:host="videos.pair2jeux.tube" />
+                <data android:host="video.splat.soy" />
+                <data android:host="videos.pofilo.fr" />
+                <data android:host="videos.pueseso.club" />
+                <data android:host="videos.side-ways.net" />
+                <data android:host="videos.squat.net" />
+                <data android:host="videos.tcit.fr" />
+                <data android:host="video.subak.ovh" />
+                <data android:host="videos.ubuntu-paris.org" />
+                <data android:host="videos.uni-corn.me" />
+                <data android:host="videos.upr.fr" />
+                <data android:host="videos.wakapo.com" />
+                <data android:host="video.taboulisme.com" />
+                <data android:host="videotape.me" />
+                <data android:host="video.tedomum.net" />
+                <data android:host="video.turbo.chat" />
+                <data android:host="video.typica.us" />
+                <data android:host="video.valme.io" />
+                <data android:host="video.vny.fr" />
+                <data android:host="video.wivodaim.net" />
+                <data android:host="video.writeas.org" />
+                <data android:host="video.yukari.moe" />
+                <data android:host="vid.lelux.fi" />
+                <data android:host="vid.lubar.me" />
+                <data android:host="vid.ncrypt.at" />
+                <data android:host="vid.y-y.li" />
+                <data android:host="vidz.dou.bet" />
+                <data android:host="viid.ga" />
+                <data android:host="vis.ion.ovh" />
+                <data android:host="v.kretschmann.social" />
+                <data android:host="v.lastorder.xyz" />
+                <data android:host="v.lesterpig.com" />
+                <data android:host="vloggers.social" />
+                <data android:host="v.mbius.io" />
+                <data android:host="vod.ksite.de" />
+                <data android:host="vod.mochi.academy" />
+                <data android:host="watch.44con.com" />
+                <data android:host="watch.snoot.tube" />
+                <data android:host="wetube.ojamajo.moe" />
+                <data android:host="widemus.de" />
+                <data android:host="www.videos-libr.es" />
+                <data android:host="www.yiny.org" />
+                <data android:host="xxxporn.co.uk" />
```

Comment:
@wb9688 For example because of this. We just decided to not add support for a pornsite. But now we support opening porn URLs.

Reply:
However it was already possible to add that instance to NewPipe, and those URLs aren't displayed anywhere in NewPipe itself anyway.

Reply:
There are many others that are porn sites as well, such as doby.io. Literally the third one I checked randomly.

Reply:
By adding native support it is different than allowing users to add their own.

Reply:
@gkeegan: The only thing this adds is when you go to e.g. doby.io in your browser, it will allow you to open it in NewPipe. We aren't adding it as a default instance or whatever. Also note that it isn't possible to dynamically add URLs to the list of sites the app can open.

Reply:
It also adds it to the "supported links" page, which is going to be clogged with a bunch of sites that nobody cares about, and also will have porn sites in it. 

NewPipe is not and should not ever endorse such content by adding any native support for it or references in the code, either by adding link support or by adding as a service.

Reply:
Okay. What do we do now? Which instances should we add to the manifest?
two suggestions:
- take the ~10 most popular
- add the ones @B0pol suggested (could you list them again, please?)

Apart from that we should put a note somewhere that we are willing to add support for opening links from more instances on request as long as they match our inclusion criteria.

Reply:
IMO:
+ most popular, definitely
+ the ones used for tests
+ user wanted instances, updated from time to time.

and you'll need to update [inclusion criteria](https://teamnewpipe.github.io/documentation/00_Prepare_everything/) because technically from this, this instance could be added.

To add instance, we should check mainly check if videos are marked as NSFW to avoid such cases(It should be blurred).
I'll push tomorrow a first list.

Reply:
Which kid is going to check the "supported links" list, of all things? I don't get the brouhaha around supporting Peertube links with adult content in them. We literally have a toggle that allows us to watch age restricted content on Youtube.

Newpipe shouldn't become a censor, IMO. Newpipe doesn't show adult content by default, which is why the toggle is off when you first open the app. The user has the freedom to enable it.

Which should be the case with Peertube as well. Discriminating between instances like this is unnecessary. As @wb9688 said, this just allows links from a site to be opened in the app, nothing more. You have to first go to the site and click on a video, at which point, who exactly are we "protecting"?

Anyway, the instances list has a toggle called "Sensitive videos" which can be set to Hide. This only shows instances that completely hide adult content by default. Maybe that list can be fetched programmatically.

Reply:
The difference between YouTube's "restricted content" and the content on these sites is VERY large. I agree NewPipe shouldn't be a filter, but if someone wants to watch it, they can add it in the instances list, but we shouldn't add it into the code for link support and thus endorse it.

Personally this is less about protecting people and more about the ethics of supporting it natively. The industry preys on people who are vulnerable and then chews them up and spits them out broken and addicted to drugs. I want no part of a project that has anything to do with that.

Reply:
Adding it to the list of instances without link support would mean that the user could open the site but not share links to Newpipe, which could generate bug reports, since this behaviour would be different from YT, SC and MediaCCC.

Regarding the ethics part: Newpipe, being just a client, has nothing to do with the websites it fetches from. It doesn't give them ad revenue, or any other kind of benefit. In fact, you could argue that since it accesses their servers without giving anything back, it's kind of a drain on their resources, and hence their money, hurting them economically, on however small a scale. :P

Anyway, I don't think it really matters on a practical level. I've never even heard of anyone accessing Peertube to watch porn. Philosophically, I'm against censorship.

Reply:
@opusforlife2 I agree with you about censorship. We are not in a position - either ethically or scientifically - to decide which content is good or bad, true or false, to be looked at or preferred, or not to be viewed. I am aware of the problematic that arises when opening not all instances is supported or worse, when some instances are deliberately blocked. Pornography was only one example, which immediately caught my eye by the URL. The question is whether we simply support opening all instances. We are in no way responsible for their content and users should be aware of it because they already got the URL.

They main problem I see, when adding all 449 instances, is crashing old devices, or at least create lags. Every shared URL would be checked against 449 patterns for PeerTube and some others for YouTube (+ invidio + hooktube), SoundCloud, mediaCCC. And that's only for NewPipe. I should have communicated that clearer.
For this reason, I suggested to only add a few instances to the manifest. We can add more on request or even ask our users in a blog post for their most used instances. Adding the most popular for now does not prevent us from adding unknown or small instances later.

Reply:
More clarification: now, since PeerTube was added, since more than two years for other links, and for future not-in-manifest instances, you can still open a link with NewPipe by sharing to it. Then the linkHandlers will try to open or say Unsupported URL if NewPipe can't open.

The difference is: if it's in manifest, you will be able to set NewPipe to automatically open this link, without needing to share it to NewPipe.

Reply:
Well, that makes sense. Technical reasons supersede everything (well, mostly) so that is probably for the best.

> The difference is: if it's in manifest, you will be able to set NewPipe to automatically open this link, without needing to share it to NewPipe.

Oh. I see. I thought it would affect the share functionality. That clears it up.